src/backend/access/external/libchurl.c

```diff
@@ -312,6 +312,14 @@ CHURL_HANDLE churl_init_upload(const char* url, CHURL_HEADERS headers)
 	context->upload = true;
 	clear_error_buffer(context);
 
+	/* needed to resolve pxf service address */
+	struct curl_slist *resolve_hosts = NULL;
+	char *pxf_host_entry = (char *) palloc0(strlen(pxf_service_address) + strlen(LocalhostIpV4Entry) + 1);
+	strcat(pxf_host_entry, pxf_service_address);
```

Comment:
This step might be unnecssary if the pxf_service_address itself is based on the IP address